app/controllers/hiring_staff/vacancies/job_summary_controller.rb

```diff
@@ -0,0 +1,34 @@
+class HiringStaff::Vacancies::JobSummaryController < HiringStaff::Vacancies::ApplicationController
+  before_action :redirect_unless_vacancy
+
+  def show
+    @job_summary_form = JobSummaryForm.new(@vacancy.attributes)
+  end
+
+  def update
+    @job_summary_form = JobSummaryForm.new(job_summary_form_params)
+
+    if @job_summary_form.valid?
+      update_vacancy(job_summary_form_params, @vacancy)
+      update_google_index(@vacancy) if @vacancy.listed?
+      return redirect_to_next_step_if_save_and_continue
+    end
+
+    render :show
+  end
+
+  private
+
+  def job_summary_form_params
+    (params[:job_summary_form] || params).permit(:job_summary, :about_school)
+                                         .merge(completed_step: current_step)
+  end
+
+  def redirect_to_next_step_if_save_and_continue
+    if params[:commit] == I18n.t('buttons.save_and_continue')
+      redirect_to school_job_review_path(@vacancy.id)
+    elsif params[:commit] == I18n.t('buttons.update_job')
+      redirect_to edit_school_job_path(@vacancy.id), success: I18n.t('messages.jobs.updated')
+    end
```

Comment:
is this safe if neither condition is met?

Reply:
If we changed the way forms are submitted this would have to change, but unless that happens I think it's safe to assume one of these will be true